### ISSUE TITLE
Search feature

### ISSUE THREAD

**Issue by simevidas** [simevidas/webplatformdaily-site#101]
### Current limitations

 - only tested in Firefox and Chrome on desktop using keyboard (no touch support yet)
 - the search algorithm is (for now) limited to finding exact matches in the Markdown source 

### How to use

1. on the site, press <kbd>/</kbd> (slash key) to activate and focus the search field
2. enter a search term ([see this YouTube video for examples]()) and press <kbd>Enter</kbd> to execute search
3. press <kbd>Esc</kbd> to close the search interface

**Comment by miclf** [comment]
In my browser (Firefox 42.0 on OS X 10.11.1), pressing `/` actually open the browser’s built-in quick search feature (like it’s supposed to do). It seems that, in this case, the script is not able to catch the keystroke or to prevent the browser’s default behaviour.

**Comment by simevidas** [comment]
@miclf https://github.com/simevidas/webplatformdaily-site/issues/103

**Comment by brandonaaskov** [comment]
I love this feature, but I had to look up the link to this issue to find the shortcut (I kept trying "t" a la github's file finder). Any chance you'd be open to changing the key-binding?

**Comment by simevidas** [comment]
@brandonaaskov https://github.com/simevidas/webplatformdaily-site/issues/104

**Comment by rolfnl** [comment]
This is really nice... perhaps you can put the possible filters/sections at the top of the page, right now I need to scroll first to see what the sections are so I can use them as filters.. having a key handler is all for fast access, if I then have to remember the filters... I'm kind of wasting time scrolling. Plus, the sections could also serve as a filter for the content below, jumping to it or something like that.

But in any case, I like how simple everything is.

**Comment by simevidas** [comment]
@rolfnl Yeah, I should probably show a tooltip with the possible values for the `in:` filter. I haven’t defined the new UI yet, but there will definitely be a handy box explaining how to use the search in more detail. Not sure what you mean by that last sentence.
